.github/workflows/release-build.yml

```diff
@@ -2,233 +2,126 @@ name: release-build
 
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v5
+      - uses: actions/cache@v4
         with:
-          python-version: 3.10.17
-
-      - name: Install system dependencies
-        run: |
-          sudo apt-get -qq update
-          sudo apt-get install -y ca-certificates libopenblas-dev gcc-9 g++-9
-          sudo update-alternatives \
-            --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 \
-            --slave /usr/bin/g++ g++ /usr/bin/g++-9
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
 
-      - name: Build and install aihwkit wheel
+      - name: Cleanup runner
         run: |
-          pip install -r requirements.txt
-          make build_inplace
+          pip cache purge
+          sudo apt-get clean
+          sudo rm -rf /var/lib/apt/lists/*
 
-      - name: Run pytest
-        run: |
-          pip install -r requirements-dev.txt
-          make pytest
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.8.1
         env:
-          TEST_DATASET: true
-
-  # ────────────────────────────
-  # Lint on Python 3.10
-  # ────────────────────────────
-  lint-py310:
-    name: Lint (Python 3.10)
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v5
+          CIBW_ENVIRONMENT: "TORCH_VERSION_SPECIFIER='==2.3.1' MAKEFLAGS='-j4'"
+          CIBW_BEFORE_BUILD: "pip install torch==2.3.1 torchvision -f https://download.pytorch.org/whl/torch_stable.html && pip install -r requirements.txt"
+          CIBW_MANYLINUX_X86_64_IMAGE: "aihwkit/manylinux2014_x86_64_aihwkit"
+          CIBW_REPAIR_WHEEL_COMMAND: "auditwheel repair -w {dest_dir} {wheel} --exclude libtorch_python.so"
+          CIBW_BUILD: "cp38-manylinux_x86_64 cp39-manylinux_x86_64 cp310-manylinux_x86_64"
         with:
-          python-version: 3.10.17
-
-      - name: Install system dependencies
-        run: |
-          sudo apt-get -qq update
-          sudo apt-get install -y ca-certificates libopenblas-dev gcc-9 g++-9
-          sudo update-alternatives \
-            --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 \
-            --slave /usr/bin/g++ g++ /usr/bin/g++-9
+          output-dir: wheelhouse
 
-      - name: Build and install aihwkit wheel
+      - name: Sync wheels to IBM COS
+        env:
+          IBM_CLOUD_REGION: ${{ secrets.IBM_CLOUD_REGION }}
+          IBM_CLOUD_API_KEY: ${{ secrets.IBM_CLOUD_API_KEY }}
+          IBM_CLOUD_RESOURCE_GROUP: ${{ secrets.IBM_CLOUD_RESOURCE_GROUP }}
+          COS_BUCKET: ${{ secrets.COS_BUCKET }}
         run: |
-          pip install -r requirements.txt
-          make build_inplace
-
-      - name: Run lint checks
+          curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+          ibmcloud plugin install cos
+          ibmcloud login --apikey "$IBM_CLOUD_API_KEY" -r "$IBM_CLOUD_REGION" -g "$IBM_CLOUD_RESOURCE_GROUP"
```

Comment:
Not necessarily for this PR, but we have an [official github action](https://github.com/IBM/actions-ibmcloud-cli) for setting up the CLI, logging in and installing plugins.